pkg/dashboard/alert/alertStateManager.go

```diff
@@ -59,9 +65,24 @@ func (a AlertStateManager) evaluatePendingAlerts() {
 			}
 		}
 		if t != nil && t.Reached(nil, alert) {
-			a.notifManager.Broadcast(&notifications.AlertMessage{
-				Alert: *alert,
-			})
+			msg := &notifications.AlertMessage{
+				Alert: api.Alert{
+					ObjectName:    alert.ObjectName,
+					Status:        model.FIRING,
+					Type:          alert.Type,
+					Text:          t.Text(),
+					DeploymentUrl: alert.DeploymentUrl,
+				},
+			}
+
+			if a.notificationsToken != "" && alert.ImChannelId != "" {
```

Comment:
This is purely NotifManager concern.
Has no place in a struct dealing with the state of the alert.